vm/src/vm.rs

```diff
@@ -944,14 +944,50 @@ impl VirtualMachine {
         }
     }
 
+    // Equivalent to check_class. Masks Attribute errors (into TypeErrors) and lets everything
+    // else go through.
+    fn _check_cls(&self, cls: &PyObjectRef, msg: String) -> PyResult {
```

Comment:
msg part need to be a closure to avoid unnecessary formatting

```suggestion
    fn _check_cls<F>(&self, cls: &PyObjectRef, msg: F) -> PyResult
    where
        F: Fn() -> String
    {
```

Reply:
This generally follows the pattern found in the `vm.new_exception_type(&self, msg: String)`. Is there any benefit in passing a closure that calls the `format!` macro itself?

Reply:
Yes, the error message is only used when conditionally error raised. But if we format the argument itself, the format is always executed regardless the condition it is required or not.

Reply:
Ah true, address that.